### PR TITLE
updates refs for yubikey fixes, error handling

### PR DESCRIPTION
**Pull request by andrewpmartinez**

### CODE REVIEW

File: ziti/enroll/enroll.go [hunk 1]
```diff
@@ -269,31 +269,31 @@ func enrollOTT(token *config.EnrollmentClaims, cfg *config.Config, caPool *x509.
 		return err
 	}
 
-	respBody, err := ioutil.ReadAll(resp.Body)
```
Comment (andrewpmartinez) [inline ziti/enroll/enroll.go]: @dovholuknf I want you to look at this section. You were in here last and updated this logic - I want to make sure this is okay with what you saw when you were in here.

Comment (andrewpmartinez) [reply ziti/enroll/enroll.go]: Using gabs to parse the JSON allows us to not depend on ziti-edge.

Comment (dovholuknf) [reply ziti/enroll/enroll.go]: > @dovholuknf I want you to look at this section. You were in here last and updated this logic - I want to make sure this is okay with what you saw when you were in here.

we chatted. quick synopsis... @andrewpmartinez is gonna look into the causemessage/cause.message and add back the one that makes sense/is used.

File: ziti/enroll/enroll.go [hunk 2]
```diff
@@ -269,31 +269,39 @@ func enrollOTT(token *config.EnrollmentClaims, cfg *config.Config, caPool *x509.
...
+	if err != nil {
+		return errors.Errorf("enroll error: %s: could not parse body: %s", resp.Status, body)
+	}
+
+	if jsonErr.Exists("error", "message") {
+		message := jsonErr.Search("error", "message").Data().(string)
+		code := jsonErr.Search("error", "code").Data().(string)
+
+		//todo: remove causeMessage support when removed from API
+		cause := "unspecified"
```
Comment (andrewpmartinez) [inline ziti/enroll/enroll.go]: Added cause handling back in here. I plan on removing `causeMessage` in netfoundry/ziti-edge#78

File: ziti/enroll/enroll.go (hunk 1 above)
Comment (andrewpmartinez) [reply ziti/enroll/enroll.go]: Updated. See comment lower down.

**Pull request merged**